Add Business services card

raja651 opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:
This PR adds the new Business card to the in app-marketplace and renders the business card for the places where business card products would be shown. 

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/Automattic/woocommerce.com/issues/19974.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Switch the WCCOM branch to the one in this PR https://github.com/Automattic/woocommerce.com/pull/20361 and rebuild the assets.
2. run `npm run wp:cli cache flush` on WCCOM branch for the extension API's to have the new fields that is required for business services card
3.  Make sure you have some business services product on your local WCCOM. you can confirm by going to https://woocommerce.test/wp-json/wccom-extensions/1.0/search?category=business-services&country=US and confirm response has some product, otherwise add `Business services` products on your local WCCOM
4. Now go back to Core branch  and In plugins/woocommerce-admin/client/marketplace/components/constants.ts, edit the MARKETPLACE_HOST const to be https://woocommerce.test
5. Go to /plugins/woocommerce and run pnpm --filter='@woocommerce/plugin-woocommerce' build
6. Go to `Business services` [tab](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=business-services&path=%2Fextensions) 
7. confirm the business services card shown according to Figma [design](https://www.figma.com/file/5jyiYtoTF2ZUxgkulPVpOh/Business-Services?node-id=3-3406&mode=dev) for desktop and mobile
8. Test around this Issue
![Screenshot 2024-05-06 at 6 23 49 PM](https://github.com/woocommerce/woocommerce/assets/11981912/8aba4cdb-e3f9-40b6-bdd9-889a5e60542c)

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

raja651 committed:
Get new fields from extensions api and pass it to card component

raja651 committed:
add business services card

raja651 committed:
fix border for category badge and move businsess service card code to a func

raja651 committed:
Merge branch 'feature/marketplace-business-services' of github.com:woocommerce/woocommerce into add/wccom-19974-business-services-card

KokkieH reviewed (changes requested):
The cards are correctly showing for Business Service products in both search and Discover/no results tabs, but there are some visual and functionality issues with the cards themselves:

It seems bigger (taller) featured images can break out of the card:

![Extensions_‹_WooCommerce_‹_woocommerce](https://github.com/woocommerce/woocommerce/assets/11873759/9bb9f366-eb86-4ce3-85c7-b00886fd9cb1)

Figma has the description section at 150px high, but currently on the cards it's 242px. Total card height in Figma 439px compared to 530px on the cards.

Figma has a 1px wide horizontal line/border added below the featured image.

Lastly, the cards aren't clickable. They should be linking back to the product page on WCCOM (which in turn redirects to the service provider).

mcliwanow commented:
> It seems bigger (taller) featured images can break out of the card:

I would suggest using image transformation params to ensure we don't load too big images: https://docs.wpvip.com/vip-file-system/image-files/transformations/ (apart from a CSS fix, as we should have a fallback in case image is not properly resized on the server).

raja651 committed:
fix height issue for taller images for business service
- and add image transformation size for bigger images

raja651 commented:
> Figma has the description section at 150px high, but currently on the cards it's 242px. Total card height in Figma 439px compared to 530px on the cards.
> 
> Figma has a 1px wide horizontal line/border added below the featured image.

Added border and fixed the height issue for image and descriptions. The description height and card is dynamically adjusted to the image size. so fixing taller image issue fixes that as well in [ba04d6a](https://github.com/woocommerce/woocommerce/pull/47183/commits/ba04d6a5452e1a54f5c4e71d4293b752d539d07f) 

> I would suggest using image transformation params to ensure we don't load too big images: https://docs.wpvip.com/vip-file-system/image-files/transformations/ (apart from a CSS fix, as we should have a fallback in case image is not properly resized on the server).

Added useEffect to look the image size and resize it if its larger than 288px that way we don't have enlarge the smaller images and blur it . Done in [ba04d6a](https://github.com/woocommerce/woocommerce/pull/47183/commits/ba04d6a5452e1a54f5c4e71d4293b752d539d07f)

> Lastly, the cards aren't clickable. They should be linking back to the product page on WCCOM (which in turn redirects to the service provider).

I thought this was already done in other tasks, i will add that part of this task before i could do that i also wanted to check if we have to fire any tracks event similar to the one on WCCOM on the card click here?

KokkieH commented:
> I thought this was already done in other tasks

It was working before, but that's because these were using regular product cards, which this PR replaces with the new Business Services cards :)

mcliwanow commented:
> wanted to check if we have to fire any tracks event similar to the one on WCCOM on the card click here?

We should fire the same event as we do for other cards:

https://github.com/woocommerce/woocommerce/blob/ba04d6a5452e1a54f5c4e71d4293b752d539d07f/plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.tsx#L217-L232

raja651 committed:
add product url and tracks to business services card

raja651 commented:
Added the product url to make it clickable and added tracks event to Business card in [db6f13f](https://github.com/woocommerce/woocommerce/pull/47183/commits/db6f13f5e396069cb5033f8f64756789d1e89ced)

raja651 committed:
removed unnecessary useeffect and add image transformation according to card height

KokkieH commented on plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss:
```diff
@@ -284,6 +284,68 @@
+					}
+
+					p {
+						font-size: 14px;
```
```suggestion
						font-size: 13px;
```
According to Figma

KokkieH commented on plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss:
```diff
@@ -284,6 +284,68 @@
+				&__badge {
+					> span {
+						font-weight: 400;
+						font-size: 12px;
```
```suggestion
						font-size: 11px;
```

According to Figma

KokkieH commented on plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss:
```diff
@@ -284,6 +284,68 @@
+					flex-direction: column;
+					gap: 4px;
+
+					h2 {
+						margin: 0;
+					}
```
```suggestion
					h2 {
						margin: 0;
						font-size: 16px;
					}
```
According to Figma

KokkieH reviewed (approved):
Looking good. Thanks for the changes, Raja.

Just some small nitpicks about font sizes :)

I also confirmed that the Tracks event is being triggered. For other testers, run `localStorage.setItem( 'debug', 'wc-admin:*' );` in the console, then view localStorage and Cmd-Click a card. An item should be logged there containing what would be sent as a Tracks event on a live site.

raja651 committed:
fix font sizes according to figma not keep it according to wccom business serivices card

raja651 commented:
> Just some small nitpicks about font sizes :)
Fixed font sizes in [b1f36fe](https://github.com/woocommerce/woocommerce/pull/47183/commits/b1f36fe8613e766dc72926d415859df4ab813680)
